src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -258,126 +268,78 @@ class PeerActor(
       context stop self
   }
 
+  //FIXME: As the peer chain check is part of the handshaker, it should be incorporated into the handshake
```

Comment:
maybe it can just stay here, because we have other dependency on `RLPxConnectionHandler.MessageReceived` for example here:
https://github.com/input-output-hk/etc-client/pull/189/files/d4fcce558c4cb0272c967c714e1992aa6558b905#diff-d88dc0e9076bb9b1c58875187ebcd4acR297